app/models/sample.rb

```diff
@@ -470,10 +477,31 @@ def can_be_included_in_submission?
     end
   end
 
+  def control_formatted
+    return nil if control.nil?
+
+    return 'No' if control == false
+
+    case control_type
+    when 'positive'
+      'Yes (positive)'
+    when 'negative'
+      'Yes (negative)'
+    else
+      'Yes'
+    end
+  end
+
   private
 
   def safe_to_destroy
     errors.add(:base, 'samples cannot be destroyed.')
     throw(:abort)
   end
+
+  def control_is_true_if_control_type_set
+    return unless control_type && control == false
+
```

Comment:
Finding this a little hard to juggle the logic in my head, I understand what its supposed to check (That control_type is only set for controls) but still took a while to unpick.

I think you can achieve something similar with basic rails validations:
```validates :control_type, absence: true, unless: :control?```
Although I'm not sure what the default error message for that is. You might want to provide your custom one (although it will begin with 'Control type' by default.)

Reply:
Yeah we were wondering if we could use some snazzy rails validation. Will give it a go.